### PR TITLE
Workspace compatibility

### PR DESCRIPTION
**Pull request by tiagojsag**

### CODE REVIEW

File: app/src/actions/workspace.js [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+  // TODO this is only used by vesselInfo, but the data is inside a layer
+  // review wit SkyTruth
+  dispatch({
+    type: SET_TILESET_URL, payload: tilesetUrl
+  });
+
+  dispatch(initLayers(workspace.map.layers));
+}
+
+export function processProductionWorkspace(data, dispatch) {
```
Comment (nerik) [inline app/src/actions/workspace.js]: You don't need to export this function

File: app/src/actions/workspace.js [hunk 2]
```diff
@@ -0,0 +1,135 @@
+import { LAYER_TYPES } from 'constants';
+import {
+  SET_LAYERS, SET_ZOOM, SET_CENTER, SET_INNER_TIMELINE_DATES, SET_OUTER_TIMELINE_DATES, SET_BASEMAP, SET_TILESET_URL
+} from 'actions';
+import { initLayers } from 'actions/layers';
+
+export function processNewWorkspace(data, dispatch) {
```
Comment (nerik) [inline app/src/actions/workspace.js]: You don't need to export that function

File: app/src/actions/workspace.js [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    fetch(url, {
+      headers: {
+        Authorization: `Bearer ${state.user.token}`
+      }
+    }).then(res => res.json())
+      .then(data => {
+        if ('workspace' in data) {
+          return processNewWorkspace(data, dispatch);
+        }
+        return processProductionWorkspace(data, dispatch);
```
Comment (nerik) [inline app/src/actions/workspace.js]: more like `processLegacyWorkspace` ? The new format is gonna be in production too at some point.

File: app/src/actions/workspace.js [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+      url = `${MAP_API_ENDPOINT}/v1/workspaces/${ID}`;
+    }
+
+    fetch(url, {
+      headers: {
+        Authorization: `Bearer ${state.user.token}`
+      }
+    }).then(res => res.json())
+      .then(data => {
+        if ('workspace' in data) {
```
Comment (nerik) [inline app/src/actions/workspace.js]: data.workspace !== undefined

Comment (nerik) [inline app/src/actions/workspace.js]: (the in operator can expect unexpectedly with the object's prototype chain, you should use hasOwnProperty, === undefined is a more common convention)

**Comment by nerik** [comment]
IMHO we should not duplicate action dispatches, but rather build a workspace object with the same structure, then dispatch actions at a single place. We will easily forget to add those in both places in the future.

**Comment by tiagojsag** [comment]
that workspace object with the same structure for both is the redux state container.

**Comment by nerik** [comment]
> we should not duplicate action dispatches
was my main point

**Comment by nerik** [comment]
-_-

**Pull request merged**